app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/UserServiceCEImpl.java

```diff
@@ -659,16 +659,31 @@ public Flux<User> get(MultiValueMap<String, String> params) {
         return Flux.error(new AppsmithException(AppsmithError.UNSUPPORTED_OPERATION));
     }
 
+    private Boolean validateName(String name){
+        /*
+            Regex allows for Accented characters and alphanumeric with some special characters dot (.), apostrophe ('),
+            hyphen (-) and spaces
+         */
+        Boolean isValidName = name.matches("^[A-Za-zÀ-ÖØ-öø-ÿ0-9 .'-]+$");
+        return isValidName;
+    }
+
     @Override
     public Mono<User> updateCurrentUser(final UserUpdateDTO allUpdates, ServerWebExchange exchange) {
         List<Mono<Void>> monos = new ArrayList<>();
 
         Mono<User> updatedUserMono;
         Mono<UserData> updatedUserDataMono;
+        String inputName;
 
         if (allUpdates.hasUserUpdates()) {
             final User updates = new User();
-            updates.setName(allUpdates.getName());
+            inputName = allUpdates.getName();
```

Comment:
Minor, this can be declared inside here, right? 🤔

```suggestion
            String inputName = allUpdates.getName();
```